internal/planner/type_join.go

```diff
@@ -147,45 +147,41 @@ func (n *typeIndexJoin) simpleExplain() (map[string]any, error) {
 	// Add the type attribute.
 	simpleExplainMap[joinTypeLabel] = n.joinPlan.Kind()
 
-	switch joinType := n.joinPlan.(type) {
-	case *typeJoinOne:
-		// Add the direction attribute.
-		if joinType.isSecondary {
-			simpleExplainMap[joinDirectionLabel] = joinDirectionSecondaryLabel
-		} else {
-			simpleExplainMap[joinDirectionLabel] = joinDirectionPrimaryLabel
-		}
-
+	addExplainData := func(j *invertibleTypeJoin) error {
 		// Add the attribute(s).
-		simpleExplainMap[joinRootLabel] = joinType.rootName
-		simpleExplainMap[joinSubTypeNameLabel] = joinType.subTypeName
+		simpleExplainMap[joinRootLabel] = immutable.Some(j.childSide.relFieldDef.Name)
```

Comment:
question: Instead of a string, this will now be an optional type. Does the caller that receives the result of `simpleExplain()` handle this by checking `HasValue` idk why I missed seeing that check. or does it rely on the default string value being `""` if it is missing? another question, can it be missing?

Reply:
`rootName` was also `immutable.Option`